src/debug.rs

```diff
@@ -16,31 +16,77 @@ mod native {
 }
 
 /// Prints an unsigned 32-bit int.
-pub fn print32(value: u32) {
+fn print32(value: u32) {
     unsafe { native::debug_print32(value) }
 }
 
 /// Prints an unsigned 64-bit int.
-pub fn print64(value: u64) {
+fn print64(value: u64) {
     unsafe { native::debug_print64(value) }
 }
 
 /// Prints the contents of a slice.
-pub fn print_mem(slice: &[u8]) {
+fn print_mem(slice: &[u8]) {
     unsafe { native::debug_printMem(slice.as_ptr() as *const u32, slice.len() as u32) }
 }
 
 /// Prints the contents of a slice in hexadecimal format.
-pub fn print_mem_hex(slice: &[u8]) {
+fn print_mem_hex(slice: &[u8]) {
     unsafe { native::debug_printMemHex(slice.as_ptr() as *const u32, slice.len() as u32) }
 }
 
 /// Prints the value of a storage key.
-pub fn print_storage(key: &StorageKey) {
+fn print_storage(key: &StorageKey) {
     unsafe { native::debug_printStorage(key.bytes.as_ptr() as *const u32) }
 }
 
 /// Prints the value of a storage key in hexadecimal format.
-pub fn print_storage_hex(key: &StorageKey) {
+fn print_storage_hex(key: &StorageKey) {
     unsafe { native::debug_printStorageHex(key.bytes.as_ptr() as *const u32) }
 }
+
+#[macro_export]
+macro_rules! print32 {
+    ($e:expr) => {
+        #[cfg(debug_assertions)]
+        ::ewasm_api::debug::print32($e)
```

Comment:
does this really work? i think you can only call public functions from a macro - at least if the macro is used within a different module (outside debug). or am i wrong?